PiwikTracker/PiwikTracker.swift

```diff
@@ -281,18 +283,70 @@ extension PiwikTracker {
         dimensions.append(dimension)
     }
     
+    /// Set a permanent custom dimension by value and index.
+    ///
+    /// This is a convenience alternative to set(dimension:) and calls the exact same functionality. Also, it is accessible from Objective-C.
+    ///
+    /// - Parameter value: The value for the new Custom Dimension
+    /// - Parameter forIndex: The index of the new Custom Dimension
+    @objc public func setDimension(_ value: String, forIndex index: Int) {
+        set(dimension: CustomDimension( index: index, value: value ));
+    }
+    
     /// Removes a previously set custom dimension.
     ///
     /// Use this method to remove a dimension that was set using the `set(value: String, forDimension index: Int)` method.
     ///
     /// - Parameter index: The index of the dimension.
-    public func remove(dimensionAtIndex index: Int) {
+    @objc public func remove(dimensionAtIndex index: Int) {
         dimensions = dimensions.filter({ dimension in
             dimension.index != index
         })
     }
 }
 
+
+extension PiwikTracker {
+    
+    internal func getDefaultCVars() -> [CustomVariable] {
+        let currentDevice = Device.makeCurrentDevice()
+        let app = Application.makeCurrentApplication()
+        
+        return [
+            CustomVariable( name: "Platform", value: currentDevice.platform ),
+            CustomVariable( name: "OS version", value: currentDevice.osVersion ),
+            CustomVariable( name: "App version", value: app.bundleVersion ?? "unknown" )
+        ]
+    }
+    
+    /// - Returns: A view on the Custom Variables.
+    var customVariables: ArraySlice<CustomVariable> {
+        let startIndex = useDefaultCustomVariables ? 0 : 3
+        return cvars[startIndex...]
+    }
+    
+    /// Adds a new Custom Variable.
+    ///
+    /// - Parameter name: The name of the new Custom Variable
+    /// - Parameter value: The value of the new Custom Variable
+    /// - Returns: The index of the new parameter. Note that indices start at 3 to accommodate the default Custom Variables. Also note that when default Custom Variables are turned off, the returned index is offset by 3 to what is actually sent to the Piwik API.
+    @objc @discardableResult public func addCustomVariable(_ name: String, value: String) -> Int {
+        cvars.append(CustomVariable(name: name, value: value))
+        return cvars.count
+    }
+    
+    /// Remove a previously set Custom Variable. Note that the default Custom Variables cannot be removed.
+    ///
+    /// Note that, as with any array, the index of any succeeding Custom Variables is decremented by 1.
+    ///
+    /// - Parameter index: The index that was previously returned by addCustomVariable().
+    @objc public func removeCustomVariableAtIndex(_ index: Int) {
```

Comment:
What do you think about referencing the variables by name, not by index? In this case I think the `CustomVariable` struct should bring the `index` to be referenced in the API.

```Swift
@objc public func removeCustomVariable(withName name: String) {
    cvars = cvars.filter{$0.name != name }
}
```